### PR TITLE
Add gradient border support.

### PR DESCRIPTION
**Pull request by mephisto41**
webrender add gradient border support in https://github.com/servo/webrender/pull/953. This pr add support in servo.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] These changes do not require tests because it should be covered by wpt

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15905)

### CODE REVIEW

File: components/gfx/display_list/mod.rs [hunk 1]
```diff
@@ -899,6 +896,14 @@ pub struct GradientDisplayItem {
     pub stops: Vec<GradientStop>,
 }
 
+#[derive(Clone, Deserialize, HeapSizeOf, Serialize)]
+pub struct GradientDisplayItem {
+    /// Fields common to all display items.
+    pub base: BaseDisplayItem,
+
+    pub gradient: Gradient,
```
Comment (nox) [inline components/gfx/display_list/mod.rs]: Nit: document this.

File: components/gfx/display_list/mod.rs [hunk 2]
```diff
@@ -899,6 +896,14 @@ pub struct GradientDisplayItem {
     pub stops: Vec<GradientStop>,
 }
 
+#[derive(Clone, Deserialize, HeapSizeOf, Serialize)]
+pub struct GradientDisplayItem {
+    /// Fields common to all display items.
```
Comment (nox) [inline components/gfx/display_list/mod.rs]: Nit: singular, not plural.

**Review by nox — CHANGES_REQUESTED** [review]
2 nits.

**Comment by emilio** [comment]
Any chance to add a test for this if none pass?

**Comment by mephisto41** [comment]
Looks like none test pass. I'll add a new test and also address nox's comment. Thanks.

**Comment by nox** [comment]
Shouldn't this particular test live in WPT?

**Comment by emilio** [comment]
You also need to update the manifest:

```
24.81s$ bash etc/ci/manifest_changed.sh
diff --git a/tests/wpt/mozilla/meta/MANIFEST.json b/tests/wpt/mozilla/meta/MANIFEST.json
index 29267dbb2..85c151816 100644
--- a/tests/wpt/mozilla/meta/MANIFEST.json
+++ b/tests/wpt/mozilla/meta/MANIFEST.json
@@ -20345,7 +20345,7 @@
    "support"
   ],
   "css/border-image-linear-gradient.html": [
-   "3548e2a77693afb8c61c07c74e95185c4205ac99",
+   "40ea388a8a920eaecb2d0d12f998ed60ac3b56e8",
    "reftest"
   ],
   "css/border_black_groove.html": [
```

**Comment by jdm** [comment]
The manifest can be updated using `./mach update-manifest`.

**Comment by emilio** [comment]
@bors-servo retry

 * https://github.com/servo/servo/issues/14237
